core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java

```diff
@@ -190,6 +190,7 @@
 import java.util.AbstractList;
```

Comment:
In general, especially in SqlToRelConverter.java, I think you should set more temporary variables. There's a lot of repeated `get` method calls and it makes it a bit harder to read.